sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/client.py

```diff
@@ -850,7 +853,8 @@ def _client_run(self, **kwargs):
         :rtype: bool
         """
         try:
-            if self._link.current_link_credit <= 0:
+            flow = kwargs.pop("flow", True)
```

Comment:
This feels a bit hacky - what problem is this trying to solve?

Reply:
When waiting on complete_messages() (or any message settlement) we loop on do_work() to read frames that are coming in. While we are looping, we can have transfer frames coming in off the wire if there is link_credit -- those TransferFrames we end up releasing. I removed the ability to call flow() with more link credit when it reaches 0 when waiting on a disposition so that we don't get slowed down by receiving/releasing all the messages in the queue. 

Talked offline - in order to avoid having to loop on do_work() in general I am going to try to drain=True when done receiving a message with receive_messages() so that there is no link credit to receive and the following frame will either be a Disp or a detach/attach/flow etc.